internal/cmd/issue/watch/watch.go

```diff
@@ -0,0 +1,301 @@
+
+type watchCmd struct {
+	client *jira.Client
+	users  []*jira.User
+	params *watchParams
+}
+
+func (ac *watchCmd) setIssueKey(project string) error {
+	if ac.params.key != "" {
+		return nil
+	}
+
+	var ans string
+
+	qs := &survey.Question{
+		Name:     "key",
+		Prompt:   &survey.Input{Message: "Issue key"},
+		Validate: survey.Required,
+	}
+	if err := survey.Ask([]*survey.Question{qs}, &ans); err != nil {
+		return err
+	}
+	ac.params.key = cmdutil.GetJiraIssueKey(project, ans)
+
+	return nil
+}
+
+func (ac *watchCmd) setWatcher(project string) error {
+	if ac.params.user != "" && len(ac.users) == 1 {
+		ac.params.user = getQueryableName(ac.users[0].Name, ac.users[0].DisplayName)
+		return nil
+	}
+
+	var (
+		ans  string
+		last bool
+	)
+	if ac.params.user != "" && len(ac.users) > 0 {
+		last = true
+	}
+
+	for {
+		qs := &survey.Question{
+			Name: "user",
+			Prompt: &survey.Select{
+				Message: "User to add to watchers:",
+				Help:    "Can't find the user? Select search and look for a keyword or cancel to abort",
+				Options: ac.getOptions(last),
+			},
+			Validate: func(val interface{}) error {
+				errInvalidSelection := fmt.Errorf("invalid selection")
+
+				ans, ok := val.(core.OptionAnswer)
+				if !ok {
+					return errInvalidSelection
+				}
+				if ans.Value == "" || ans.Value == lineBreak {
+					return errInvalidSelection
+				}
+
+				return nil
+			},
+		}
+
+		if err := survey.Ask([]*survey.Question{qs}, &ans); err != nil {
+			return err
+		}
+		if ans == optionCancel {
+			cmdutil.Fail("Action aborted")
+			os.Exit(0)
+		}
+		if ans != optionSearch {
+			break
+		}
+		if err := ac.getSearchKeyword(); err != nil {
+			return err
+		}
+		if err := ac.searchAndWatchUser(project); err != nil {
+			return err
+		}
+		last = true
+	}
+	ac.params.user = ans
+
+	return nil
+}
+
+func (ac *watchCmd) getOptions(last bool) []string {
+	var validUsers []string
+
+	for _, t := range ac.users {
+		if t.Active {
+			name := t.DisplayName
+			if t.Name != "" {
+				name += fmt.Sprintf(" (%s)", t.Name)
+			}
+			validUsers = append(validUsers, name)
+		}
+	}
+	options := []string{optionSearch}
+
+	if last {
+		options = append(options, validUsers...)
+		options = append(options, lineBreak)
+	} else {
+		options = append(options, lineBreak)
+		options = append(options, validUsers...)
+	}
+
+	return options
+}
+
+func (ac *watchCmd) getSearchKeyword() error {
+	qs := &survey.Question{
+		Name: "user",
+		Prompt: &survey.Input{
+			Message: "Search user:",
+			Help:    "Type user email or display name to search for a user",
+		},
+		Validate: func(val interface{}) error {
+			errInvalidKeyword := fmt.Errorf("enter atleast 3 characters to search")
+
+			str, ok := val.(string)
+			if !ok {
+				return errInvalidKeyword
+			}
+			if len(str) < 3 {
+				return errInvalidKeyword
+			}
+
+			return nil
+		},
+	}
+	return survey.Ask([]*survey.Question{qs}, &ac.params.user)
+}
+
+func (ac *watchCmd) searchAndWatchUser(project string) error {
```

Comment:
Nitpick: This method is not actually watching the user and I find the name bit confusing. Let's call it `searchAndSetUser`?